src/server/qgswmsserver.cpp

```diff
@@ -980,40 +976,33 @@ QImage* QgsWmsServer::getLegendGraphics()
 }
 
 
-void QgsWmsServer::runHitTest( QPainter* painter, HitTest& hitTest )
+void QgsWmsServer::runHitTest( const QgsMapSettings& mapSettings, QPainter* painter, HitTest& hitTest )
 {
   QPaintDevice* thePaintDevice = painter->device();
 
-  // setup QgsRenderContext in the same way as QgsMapRenderer does
-  QgsRenderContext context;
+  // setup QgsRenderContext in the same way as the renderer does
+  QgsRenderContext context = QgsRenderContext::fromMapSettings( mapSettings );
   context.setPainter( painter ); // we are not going to draw anything, but we still need a working painter
-  context.setRenderingStopped( false );
-  context.setRasterScaleFactor(( thePaintDevice->logicalDpiX() + thePaintDevice->logicalDpiY() ) / 2.0 / mMapRenderer->outputDpi() );
-  context.setScaleFactor( mMapRenderer->outputUnits() == QgsMapRenderer::Millimeters ? mMapRenderer->outputDpi() / 25.4 : 1.0 );
-  context.setRendererScale( mMapRenderer->scale() );
-  context.setMapToPixel( *mMapRenderer->coordinateTransform() );
-  context.setExtent( mMapRenderer->extent() );
+  context.setRasterScaleFactor(( thePaintDevice->logicalDpiX() + thePaintDevice->logicalDpiY() ) / 2.0 / mapSettings.outputDpi() );
```

Comment:
It should not be necessary to set "raster scale factor" anymore and it should always stay equal to 1. It is a legacy used with QgsMapRenderer.

Next step would be to actually remove it from QgsRenderContext finally (in some other PR).